application/controllers/LSBaseController.php

```diff
@@ -55,6 +55,66 @@ protected function _init()
         Yii::setPathOfAlias('lsadminmodules', Yii::app()->getConfig('lsadminmodulesrootdir'));
     }
 
+    /**
+     * Validate params used for sid, gid and qid. Validate if qid is included in gid is included in sid.
+     * Validate if final survey exist
+     * Currently : used by QuestionEditorController
+     * @param integer|mixed $sid : sid param to be controlled. If invalid (not an integer) : throw a 404
+     * @param integer|mixed $gid : gid param to be controlled. If invalid (not an integer) : throw a 404
+     * @param integer|mixed $qid : qid param to be controlled. If invalid (not an integer) : throw a 404
+     * @Throw CHttpException
+     * @return false|integer the existing sid related to current params.
+     */
+    protected function getValidatedSurveyId($sid = null, $gid = null, $qid = null)
+    {
+        $oQuestion = null;
+        if ($qid) {
+            $oQuestion = Question::model()->findByPk($qid);
+            if(!$oQuestion) {
+                throw new CHttpException(404);
+            }
+        }
+        $oGroup = null;
+        if ($gid) {
+            $oGroup = QuestionGroup::model()->findByPk($gid);
+            if(!$oGroup) {
+                throw new CHttpException(404);
+            }
+            if ($oQuestion && $gid != $oQuestion->gid) {
+                // Try to hack : 400
+                throw new CHttpException(400);
+            }
+        }
+
+        $surveyId = false;
+        if(is_null($sid)) {
+            /* @todo : rempove this to move to params of function */
+            $sid = App()->request->getParam('sid', App()->request->getParam('surveyid'));
+        }
+        if ($sid) {
+            $oSurvey = Survey::model()->findByPk($sid);
+            if (!$oSurvey) {
+                throw new CHttpException(404);
+            }
+            if ($oQuestion && $sid != $oQuestion->sid) {
+                // Try to hack : 400
+                throw new CHttpException(400);
+            }
+            if ($oGroup && $sid != $oGroup->sid) {
+                // Try to hack : 400
+                throw new CHttpException(400);
+            }
```

Comment:
Does not check permissions?

Reply:
No, you ask to remove Permission checker on the global system …

> Not too happy about this, as I've expressed in our chat. Can lead to false sense of security if a controller uses another name than the ones listed in checkparams.

The only way to disable usage of a different surveyid/sid/survyId … is to get it by another function …

Then : you need the real survey id , and check permission after … the function return the sid according to qid or gid.

Reply:
Yeah, I think it's OK since it's not global anymore, more explicit. Thank you, Denis. Enjoy your vacation. :)

Reply:
In my opinion: there still an issue with this https://github.com/LimeSurvey/LimeSurvey/pull/1479/files#diff-44c5279b74ca1be33751c347b23e1585R91

Else about control on Permission : currently, need only read, but when add a function to update : need 'update' etc … Persmission check done in each function (and must be only in public function)